mindsdb/integrations/handlers/ignite_handler/ignite_handler.py

```diff
@@ -0,0 +1,238 @@
+
+    def disconnect(self):
+        """
+        Close any existing connections.
+        """
+
+        if self.is_connected is False:
+            return
+
+        self.client.close()
+        self.is_connected = False
+        return self.is_connected
+
+    def check_connection(self) -> StatusResponse:
+        """
+        Check connection to the handler.
+        Returns:
+            HandlerStatusResponse
+        """
+
+        response = StatusResponse(False)
+        need_to_close = self.is_connected is False
+
+        try:
+            self.connect()
+            response.success = True
+        except Exception as e:
+            log.logger.error(f'Error connecting to Apache Ignite!')
+            response.error_message = str(e)
+        finally:
+            if response.success is True and need_to_close:
+                self.disconnect()
+            if response.success is False and self.is_connected is True:
+                self.is_connected = False
+
+        return response
+
+    def native_query(self, query: str) -> StatusResponse:
+        """
+        Receive raw query and act upon it somehow.
+        Args:
+            query (str): query in native format
+        Returns:
+            HandlerResponse
+        """
+
+        need_to_close = self.is_connected is False
+
+        client, connection = self.connect()
+
+        try:
+            with connection:
+                with client.sql(query, include_field_names=True, schema=self.connection_data['schema']) as cursor:
+                    result = list(cursor)
+                    if result and result[0][0] != 'UPDATED':
+                        response = Response(
+                            RESPONSE_TYPE.TABLE,
+                            data_frame=pd.DataFrame(
+                                result[1:],
+                                columns=[result[0]]
+                            )
+                        )
+                    else:
+                        response = Response(RESPONSE_TYPE.OK)
+        except Exception as e:
+            log.logger.error(f'Error running query: {query} on Apache Ignite!')
+            response = Response(
+                RESPONSE_TYPE.ERROR,
+                error_message=str(e)
+            )
+
+        cursor.close()
+        if need_to_close is True:
+            self.disconnect()
+
+        return response
+
+    def query(self, query: ASTNode) -> StatusResponse:
+        """
+        Receive query as AST (abstract syntax tree) and act upon it somehow.
+        Args:
+            query (ASTNode): sql query represented as AST. May be any kind
+                of query: SELECT, INTSERT, DELETE, etc
+        Returns:
+            HandlerResponse
+        """
+
+        # if isinstance(query, ASTNode):
+        #     query_str = query.to_string()
+        # else:
+        #     query_str = str(query)
+        #
+        # return self.native_query(query_str)
+        renderer = SqlalchemyRender('ignite')
```

Comment:
@MinuraPunchihewa Are you sure this will work? Ignite is not supported dialect in the Renderer https://github.com/mindsdb/mindsdb_sql/blob/cac792efdc13762d8ab73e93006abd7b6dd5bafc/mindsdb_sql/render/sqlalchemy_render.py#L19

Reply:
@ZoranPandovski Apologies, it looks like I have uncommented the wrong block of code when pushing this through. I have pushed a couple more commits, it should be working fine now. I just tested it again,
![image](https://user-images.githubusercontent.com/49385643/225408690-a1bc03bf-0d13-42ce-b5ff-7813cca408ca.png)